### ISSUE TITLE
MIssmatch between problem_type and loss functions in DistillBert for sequence classification

### ISSUE THREAD

**Issue by sid-sundrani** [huggingface/transformers#12014]
## Environment info
- Platform: Windows 
- Python version: 3.8
- PyTorch version (GPU?): 1.8

### Who can help
@LysandreJik

Models:
Distillbert for sequence classification

## Information
The problem_type in config is not correct with the loss function applied. Can be seen [here](https://github.com/huggingface/transformers/blob/242ec31aa59b358e631d981b545fd08330584ea8/src/transformers/models/distilbert/modeling_distilbert.py#L649) 

Eg. for the problem type "single_label_classification", the CrossEntropyLoss is used, and the logic to detect "single_label_classification" indicates there should be multiple classes. So its just that the labels of problem_type are switched. Hence, if someone uses actually uses this config of probelm_type to choose  their task (binary, or multiclass), the applied loss would be inccorect 

## Expected behavior
For problem_type, the "single_label_classification", the BCE loss should be applied

**Comment by LysandreJik** [comment]
cc @abhi1thakur
